paddle/fluid/pybind/op_function.h

```diff
@@ -173,6 +211,708 @@ static inline void HandleViewBetweenInputAndOutput(
             << "), share allocation and inplace version.";
   }
 }
+
+extern PyTypeObject* g_VarBase_PyType;
+extern PyTypeObject* g_VarType_PyType;
+
+inline bool PyObject_CheckBool(PyObject* obj) { return PyBool_Check(obj); }
+
+inline bool PyObject_CheckLong(PyObject* obj) {
+  return (PyLong_Check(obj) && !PyBool_Check(obj)) ||
+         PyObject_IsInstance(obj, (PyObject*)g_VarType_PyType) ||  // NOLINT
+         (_import_array() >= 0 && PyArray_IsScalar((obj), Integer)) ||
+         PyObject_IsInstance(obj, (PyObject*)g_VarBase_PyType);  // NOLINT
+}
+
+inline bool PyObject_CheckFloat(PyObject* obj) {
+  return PyFloat_Check(obj) || PyLong_Check(obj) ||
+         (_import_array() >= 0 && PyArray_IsScalar(obj, Floating)) ||
+         (_import_array() >= 0 && PyArray_IsScalar(obj, Integer)) ||
+         PyObject_IsInstance(obj, (PyObject*)g_VarBase_PyType);  // NOLINT
+}
+
+inline bool PyObject_CheckString(PyObject* obj) { return PyUnicode_Check(obj); }
+
+static inline void ConstructAttrMapFromPyArgs(
+    const std::string& op_type, PyObject* args, ssize_t attr_start,
+    ssize_t attr_end, paddle::framework::AttributeMap& attrs) {  // NOLINT
+  PADDLE_ENFORCE_EQ(
+      (attr_end - attr_start + 1) % 2, 0,
```

Comment:
Not important. I suggest passing attr_end+1 directly to avoid some problems, [start, end) is commonly used (which means start is reachable and end is not).

Reply:
done，thx！